src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -591,7 +591,32 @@ private function getAdjacentRecordID($offset)
 
         $map = $list->limit($limit, $limitOffset)->column('ID');
         $index = array_search($this->record->ID, $map ?? []);
-        return isset($map[$index+$offset]) ? $map[$index+$offset] : false;
+        $position = $index + $offset;
+        $page = $currentPage;
+        $lastPosition = count($map) - 1;
+        $hasMorePages = $this->getNumPages($gridField) > $currentPage;
+
+        if ($position === 0 && $currentPage > 1) {
+            $page = $currentPage - 1;
+        } elseif ($position === $lastPosition && $hasMorePages) {
+            $page = $currentPage + 1;
+        }
+        $gridField->State->GridFieldPaginator->currentPage = (int)$page;
+        
+        return isset($map[$position]) ? $map[$position] : false;
+    }
+
+    /**
+     * Gets the number of GridField pages
+     * @return int
+     */
+    public function getNumPages(GridField $gridField)
```

Comment:
```suggestion
     */
    public function getNumPages(GridField $gridField): int
```
Prefer typehints over phpdocs for new methods.

Reply:
I would prefer to keep consistency in how we document a return type in this class.
We have @return * everywhere.

Reply:
Would be better to create task for refactoring phpdocs to typehints.

Reply:
The _existing_ phpdocs will be refactored when we strongly type everything, which there is a spike for currently: https://github.com/silverstripe/silverstripe-framework/issues/10414

In the meantime, the standard practice we've had is to use typehints and _not_ phpdocs for all _new_ API.

Reply:
DONE